torch/_inductor/lowering.py

```diff
@@ -2067,13 +2067,35 @@ def require_channels_last(_, *args, **kwargs):
     return args, kwargs
 
 
-def constrain_to_fx_strides(fx_node, *args, **kwargs):
+def constrain_to_fx_strides(fx_node, *args, ignore_mutated_args_FIXME=False, **kwargs):
     def apply_constraint(arg, fx_arg):
         if isinstance(arg, ir.IRNode):
             stride_order = ir.get_stride_order(fx_arg.meta["val"].stride())
             return ir.ExternKernel.require_stride_order(arg, stride_order)
         return arg
 
+    if ignore_mutated_args_FIXME:
+        assert isinstance(fx_node.target, torch._ops.OpOverload)
+        schema = fx_node.target._schema
+
+        new_args = []
+        new_kwargs = {}
+        schema_args, schema_kwargs = torch._library.utils.schema_args_kwargs(schema)
+        for arg, fx_arg, schema_arg in zip(args, fx_node.args, schema_args):
```

Comment:
There is no contract that the fx node at this point is using `fx_node.args` instead of `fx_node.kwargs`. What I would expect to see here is `torch.fx.operator_schemas.normalize_function` and then iterate over the schema by argname

Reply:
Sounds good

Reply:
By the way... the code as previously written assumes that fx_node.{args, kwargs} are zip-able with {args, kwargs}. Are you saying that's incorrect too?

Reply:
I updated the code to handle the case where the schema_args/schema_kwargs can be different from args/kwargs. The way constrain_to_fx_strides is called (and from how the code was written previously) it looks like fx_node.{args, kwargs} being zip-able with {args, kwargs} is a reasonable assumption (but if we think that's a bug I can fix it in a follow-up -- I want this PR to go in without changing too much since it needs to be cherry-picked)